### ISSUE TITLE
Dependencies with non-overlapping compiler specifications are refused

### ISSUE THREAD

**Issue by p0nce** [D-Programming-Language/dub#630]
**Problem**

I can't use the following form:

```
{
    "name": "lol",            

    "dependencies-ldc": {
        "logger": "~>0.3.0"
    },
    "dependencies-gdc": {
        "logger": "~>0.3.0"
    }
}
```

**Output**
` The dependency 'logger' is specified more than once.` 
I think that no, it isn't.

Unfortunately this is the only way I can imagine to solve this bug: https://github.com/d-gamedev-team/gfm/issues/124

**Comment by p0nce** [comment]
I guess it depends on #615.

**Comment by s-ludwig** [comment]
Yes, it's a duplicate of #615. Currently platform suffixes are simply ignored - they need to at least issue a proper warning. The workaround described there can be applied here, too.

BTW, I think ideally the `logger` package should simply have a proper `static if` that checks for the `std.experimental.logger` module and simply forwards to that using `public import`.

**Comment by p0nce** [comment]
Uh, absolutely. Going to report it there.

**Comment by p0nce** [comment]
Closed as duplicate.
